File: spring_backend/src/main/java/com/spring_backend/card/controller/dto/CardAccessResponseDTO.java
package com.spring_backend.card.controller.dto;

import com.spring_backend.card.dao.model.CardAccessType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class CardAccessResponseDTO {
    private CardAccessType accessType;
}

File: esp32_part/lib/dto/CardAccessResponseDTO.cpp
#include "CardAccessResponseDTO.h"

CardAccessResponseDTO::CardAccessResponseDTO(String accessType)
    : _accessType(accessType) {
}

CardAccessResponseDTO CardAccessResponseDTO::mapJsonToDTO(JSONVar json) {
    String accessType = (const char*)json["accessType"];
    return CardAccessResponseDTO(accessType);
}
